lib/api/playlist.js

```diff
@@ -0,0 +1,123 @@
+const ApiResource = require('./api_resource')
+const PageableApiResource = require('./pageable_api_resource')
+const qs = require('qs')
+
+/**
+ * @class Playlist
+ */
+class Playlist extends ApiResource {
+  /**
+   * Retrieves flag if playlist is available for a specific channel.
+   *
+   * @param {Number}    channelId
+   *
+   * @returns {*}
+   */
+  isEnabled (channelId) {
+    return this.context.authRequest('get', `/channels/${channelId}/settings/playlists.json`)
+      .then((res) => {
+        return Promise.resolve(res.is_enabled)
+      })
+  }
+
+  /**
+   * Lists all Playlists on an channel.
+   *
+   * @param {Number} pageSize - The number of results to show per page.
+   * @param {Number} page     - The page to retrieve.
+   * @param {boolean} includeEmptyLists   - Should empty playlists be returned
+   * @returns {Promise}
+   */
+  list (channelId, pageSize = 50, page = 1, includeEmptyLists=false) {
+    pageSize = Math.min(pageSize, 50)
+    return this.context.authRequest('get', `/channels/${channelId}/playlists.json`, qs.stringify({
+      pageSize,
+      page,
+      filter:{
+        include_empty: includeEmptyLists
+      }
+    }))
+      .then((res) => {
+        return Promise.resolve(new PageableApiResource(this.context, 'playlists', res.playlists, res.paging))
+      })
+  }
+
+  /**
+   * Lists all videos in a Playlist
+   *
+   * @param {Number} playlistId
+   * @param {Number} pageSize - The number of results to show per page.
+   * @param {Number} page     - The page to retrieve.
+   *
+   * @returns {Promise}
+   */
+  listVideos (playlistId, pageSize = 200, page = 1) {
+    pageSize = Math.min(pageSize, 200)
+    return this.context.authRequest('get', `/playlists/${playlistId}/videos.json`, qs.stringify({
+      pageSize,
+      page
+    })).then((res) => {
+      return Promise.resolve(new PageableApiResource(this.context, 'video', res.videos, res.paging))
+    })
+  }
+
+  /**
+   * Creates a new playlist.
+   *
+   * @param {string}    title - The title of the playlist.
+   * @param {Number}    channelId
+   * @param {Number}    is_enabled - Whether the playlist is enabled or not. Values: 1 (enabled), 0 (disabled). (1 by default)
```

Comment:
All variables should be in `camelCase`. `is_enabled` should be renamed to `isEnabled`.